### DISCUSSION TITLE
Deno ?

### DISCUSSION THREAD

**Discussion by JeromeBourgeois** [vendurehq/vendure#1660] [General]
Hi the team,

GraphQL, Remix, etc., can work on Deno, is it the case of Vendure ?

**Comment by michaelbromley** [comment]
Hi,

This is really down to whether NestJS (the underlying framework upon which Vendure is built) will support Deno. Currently the answer is **no**. You can read more discussions here: https://github.com/nestjs/nest/issues?q=is%3Aissue+sort%3Aupdated-desc+deno+is%3Aclosed

**Reply by LeaKitty** [reply]
Hi,

Let me enter the subject ^^

- If I understand correctly, REMIX is just to use for the frontend and does not replace Next entirely ?
- Another more personal question, not knowing neither React, not Vue, which one do you advise me to learn ? I'm interested in Remix features, but for the moment, it is not yet stable on Vue, and the latter has "Vue StoreFront" and Nuxt3.

**Reply by michaelbromley** [reply]
Hi @LeaKitty,
In this thread we're talking about NestJS, not to be confused with Next.js. NestJS is a backend framework, whereas Next.js is mainly a front-end framework (albeit with a bit of server-side capabilities). Remix is also comparable to Next.js - they are very similar in many ways.

As for what to learn, well React is more widely used than Vue right now, but honestly both are very good choices and I'd say beyond the larger potential job market for React skills, it's probably down to personal preference and which one clicks with you the best.

**Reply by silouanwright** [reply]
Just because I'm curious: would you ever consider using the Deno alternative, Danet? 
https://github.com/Savory/Danet

**Reply by michaelbromley** [reply]
@reywright I hadn't come across Danet before so thanks for bringing that to my attention. To answer your question: no, I would not. At least, not unless Deno becomes preferred over Node.js in general. 

I know that the Nest core team are currently not planning to work on Deno support, but I do have to wonder whether it would make more sense for the Danet devs (or others) to spend some time figuring out Nest/Deno compat rather than creating a whole new framework that basically re-creating Nest from the ground up for Deno.
